pkg/sshd/lock.go

```diff
@@ -0,0 +1,68 @@
+package sshd
+
+import (
+	"fmt"
+	"sync"
+)
+
+// RepositoryLock interface that allows the creation of a lock associated
+// with a repository name to avoid simultaneous git operations
+type RepositoryLock interface {
+	// Lock aquires a lock for a repository
+	Lock(repoName string) error
```

Comment:
@aledbf I think for now we could solve that by passing `time.Duration(0)` to the `Lock` func.

If we use a network-based lock in the future, though, we're going to need to account for network delays, failure cases, etc... and a timeout is a simple first solution for doing so.